### ISSUE TITLE
path_expand has trouble with special characters in the expanded bit under Windows (not UTF8)

### ISSUE THREAD

**Issue by famuvie** [r-lib/fs#291]
`path_expand()` handles special characters, except when they are in the expanded part. Here is a MRE in a non-UTF8 Windows 10 system.

``` r
## I have a funny user name
Sys.getenv("HOME")
#> [1] "C:\\Users\\Facundo Muñoz\\Documents"

## path_expand() understands special characters
fs::path_expand( "C:/Users/Facundo Muñoz/")
#> C:/Users/Facundo Muñoz

## but they cause trouble when expanded
fs::path_expand("~/")  # as in fs::path_home()
#> Error in base::nchar(x, type, allowNA, keepNA) : 
#>   invalid multibyte string, element 1

## this is how the function works
fs::path_expand
#> function (path) 
#> {
#>     path <- enc2utf8(path)
#>     path_tidy(.Call(fs_expand_, path, Sys.getenv("R_FS_HOME") != 
#>         "" || is_windows()))
#> }
#> <bytecode: 0x5621b217fdb0>
#> <environment: namespace:fs>

## in the first case, the string is encoded previously
## and the special character is preserved
.Call(fs:::fs_expand_, enc2utf8("Muñoz"), TRUE)
#> [1] "Muñoz"

## in the second case, the path is expanded afterwards
## and the special character is scrambled
.Call(fs:::fs_expand_, "~", TRUE)
#> "C:/Users/Facundo Mu\xf1oz"
```

**Comment by EspenRosenquist** [comment]
Is there any update on this?

This is an error that breaks the entire tidyverse for me as I cannot read files placed in paths with exiting, funny characters. usethis-functions are just dead in the water, as many uses the fs library to find paths (e.g. usethis::edit_r_environ).

The documentation of path_expand (which is the same document as for path_home_r()) it says that 

"Users can set the R_FS_HOME environment variable to override the definitions on any platform."

However, path_home_r does not look for that variable, it jumps straight to "~":

`fs::path_home_r`
#> function (...) 
#> {
#>   path(path_expand_r("~/"), ...)
#> }

I am pretty desperate now, as this - with RStudio and R base similar or compounded errors - who knows - prevents me from using RStudio on Windows (yes yes, encoding and everything has been tried to hell and back and three working days have just gone searching for SOME solution). 

```
normalizePath("~")
 [1] "C:\\Users\\userpath\\OneDrive - organisation name with�\\Dokumenter"
 Warning message:
 In normalizePath(path.expand(path), winslash, mustWork) : 
 path[1]="C:/Users/userpath/OneDrive - organisation name with�/Dokumenter": 
 The system cannot find the path specified
```

RStudio version 2022.07.2 576
R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

**Comment by jennybc** [comment]
My understanding is that the ability for the user to configure R_FS_HOME applies to `path_expand()` (not `path_expand_r()`):

https://github.com/r-lib/fs/blob/238032fd0454fa54816c5af7cff6385b1cf81d17/R/path.R#L285-L290

**Comment by EspenRosenquist** [comment]
Yes, but then path_expand_r() should perhaps not refer to "~/" internally (path_expand("~/")) or path_expand should have a clause to test for ~ when input as first character in the parameter, perhaps with a regex or simular - and handle it … correctly?

**Comment by jennybc** [comment]
I'm not attempting to address the issue here, just trying to help you maybe get to a better place, with things as they are.
